Remove the unused "GetIsPureXfa" message handler; and avoid unnecessary parsing when no structTree is available (PR 13069 follow-up, PR 13221 follow-up)

Snuffleupagus opened the pull request:
- Remove the unused "GetIsPureXfa" message handler in the worker (PR 13069 follow-up)

   Looking at the API, there's no code which actually sends this message. Most likely it's a left-over from a previous version of PR 13069, since the `isPureXfa` parameter is being included in the "GetDoc" message.

 - Avoid unnecessary parsing, in `Page.GetStructTree`, when no structTree is available (PR 13221 follow-up)

   It's obviously (a bit) more efficient to return early in `Page.getStructTree`, rather than trying to first "parse" an *empty* structTree-root.

   *Somehow I didn't think of this yesterday, but this feels like a much better solution overall; sorry about the churn here!*

Snuffleupagus committed:
Remove the unused "GetIsPureXfa" message handler in the worker (PR 13069 follow-up)

Looking at the API, there's no code which actually sends this message. Most likely it's a left-over from a previous version of PR 13069, since the `isPureXfa` parameter is being included in the "GetDoc" message.

Snuffleupagus committed:
Avoid unnecessary parsing, in `Page.GetStructTree`, when no structTree is available (PR 13221 follow-up)

It's obviously (a bit) more efficient to return early in `Page.getStructTree`, rather than trying to first "parse" an *empty* structTree-root.

*Somehow I didn't think of this yesterday, but this feels like a much better solution overall; sorry about the churn here!*

Snuffleupagus commented:
/botio test

Sagar-SalesforceTechnicalArchitech commented:
Hi Team,
I am implementing PDFJS in salesforce. Using version 2.7.5. and I am passing base64 to it as shown in example of pdfJS. I am getting error "cannot read property document of undefined" which is globalThis.document. I need a urgent help

Snuffleupagus committed:
Ensure that the data is loaded, in the "GetPageJSActions" message handler

Similar to all other data accesses, note e.g. the "GetDocJSActions" handler just above, we need to ensure that a `MissingDataException` isn't propagated to the main-thread if this data is accessed while the PDF document is still loading.

Snuffleupagus commented:
/botio integrationtest

timvandermeij commented:
Thanks!
